server/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -66,7 +66,7 @@ public void prepare() throws IOException {
     }
 
     private LogicalPlan plan(String statement) {
-        return assertMaxBytesAllocated(ByteSizeUnit.MB.toBytes(8), () -> sqlExecutor.logicalPlan(statement));
+        return assertMaxBytesAllocated(ByteSizeUnit.MB.toBytes(25), () -> sqlExecutor.logicalPlan(statement));
```

Comment:
`testSelectCountStarIsOptimizedInsideRelations` failed with:

```
java.lang.AssertionError: 
Expected: a value less than or equal to <8388608L>
     but: <24224152L> was greater than <8388608L>
	at __randomizedtesting.SeedInfo.seed([A1199D5560FEB506:3485992BF20EF71E]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at io.crate.testing.MemoryLimits.assertMaxBytesAllocated(MemoryLimits.java:46)
	at io.crate.planner.operators.LogicalPlannerTest.plan(LogicalPlannerTest.java:69)
	at io.crate.planner.operators.LogicalPlannerTest.testSelectCountStarIsOptimizedInsideRelations(LogicalPlannerTest.java:221)
```

Not sure why this increased in that particular case, because that isn't even affected by the other changes in this PR. Will follow up on this separately.